Разметка всех страниц

AndreyKaliushka opened the pull request:
---
:mortar_board: [Разметка всех страниц](https://up.htmlacademy.ru/htmlcss/35/user/2140589/tasks/5)

:boom: https://htmlacademy-htmlcss.github.io/2140589-sedona-35/2/

AndreyKaliushka committed:
Добавляет страницу каталога

AndreyKaliushka committed:
Добавляет Хеадер

AndreyKaliushka committed:
Добавляет хеадер в каталог

AndreyKaliushka committed:
Добавляет ссылку заглушку на кнопку в хеадер(?)

Он почему-то резко потребовал коммит, когда это было в предыдущем

AndreyKaliushka committed:
Добавляет футер на главную страницу

AndreyKaliushka committed:
Добавляет футер в каталог

AndreyKaliushka committed:
Добавляет разметку главной страницы

AndreyKaliushka committed:
Добавил разметку главной страницы и каталога

В общей сумме сидел 7-8 часов минимум и кажется очень много косяков:(

talequale commented on catalog.html:
```diff
@@ -0,0 +1,196 @@
+<ul class="navigation-list">
+  <li class="navigation-item">
+<a class="navigation-logo" href="index.html"><img class="logo" src="" alt="Логотип Седона"></a>
+  </li>
```
логотип является частью навигации, но вряд ли он входит в тот же список, что и остальные текстовые пункты. он неоднороден с ними. можно его просто вынести за пределы этого списка

talequale commented on catalog.html:
```diff
@@ -0,0 +1,196 @@
+  <li class="navigation-item">
+    <a class="navigation-link">Лайки<sup>12</sup></a>
+      </li><li class="navigation-item">
+       <a class="navigation-link header-button" href="#"><input class="header-button" type="button" name="ХОЧУ СЮДА" value="ХОЧУ СЮДА!"></a>
```
тег input не может размещаться внутри тега `a`, вообще никакие интерактивные элементы не могут лежать внутри ссылки. к тому же для разметки кнопок используется не тег input, а тег button (который тоже не может лежать в ссылке)

другой момент - не всё, что выглядит визуально как кнопка, обязательно является кнопкой, то есть не обязательно размечается тегом button. "хочу сюда" вполне можно разметить ссылкой с классом, отличающимся от navigation-link. трудно себе представить, что делает кнопка "хочу сюда", скорее это всё-таки переход на другую страницу

talequale commented on index.html:
```diff
@@ -5,8 +5,153 @@
+</header>
+    <main class="main-index">
+<h1 class="visually-hidden">Седона</h1>
+<img class="main-logo" src="" alt="Логотип Седона">
```
эти заголовок и картинку надо обернуть в div или section, потому что это отдельный блок со своими стилями (размеры, фон, декоративная маска в виде треугольников и т д)

talequale commented on index.html:
```diff
@@ -5,8 +5,153 @@
+      <h3 class="advantages-title">Небольшая площадь</h3>
+      <p class="advantages-description">Все достопримечательности находятся очень близко</p>
+    </li>
+  </ul>
```
в чём смысл списка, в котором один элемент? 
весь вот этот блок - список
![HTML 1 : Седона – Figma 2022-09-26 22-03-05](https://user-images.githubusercontent.com/11165551/192358960-0c94a012-cc71-4ff9-b4a2-ec9ad0acd856.jpg)

talequale commented on index.html:
```diff
@@ -5,8 +5,153 @@
+      <p class="advantages-description">Седона — не аттракцион для туристов, там течёт своя жизнь</p>
+    </li>
+  </ul>
+  <img class="cafe-picture" src="" alt="Фотография кафе">
```
картинка относится к первому пункту списка, который про "Настоящий городок"

talequale commented on index.html:
```diff
@@ -5,8 +5,153 @@
+      <p class="advantages-description">Большинство едет в Гранд Каньон и толпится там</p>
+    </li>
+  </ul>
+  <img class="bridge-picture" src="" alt="Фотография моста дьявола">
```
картинка относится к блоку "Мало туристов"

talequale commented on index.html:
```diff
@@ -5,8 +5,153 @@
+<h3 class="interested-title">Заинтересовались?</h3>
+<p class="interested-description">Укажите предполагаемые даты поездки, и мы покажем вам лучшие предложения гостиниц в Седоне</p>
+  <a class="interested-button" href="catalog.html"> <input class="interested-button" type="button" name="ПОИСК ГОСТИНИЦЫ В седоне" value="ПОИСК ГОСТИНИЦЫ В седоне"></a>
+          </li>
```
это не input в ссылке, это либо просто тег button, либо просто ссылка

talequale commented on index.html:
```diff
@@ -5,8 +5,153 @@
+<input type="text" name="email" placeholder="Ваш e-mail">
+<button class="distribution-button" type="submit" name="button" value="Подписаться">Подписаться</button>
+</form>
+</footer>
```
вряд ли это футер, просто блок рассылки, можно разметить тегом section

talequale commented on index.html:
```diff
@@ -5,8 +5,153 @@
+<footer class="distribution">
+<h3 class="distribution-title">Подпишитесь на рассылку</h3>
+<p class="distribution-description">Только полезная информация и никакого спама, честное бойскаутское!</p>
+<form class="distribution-email" action="https://echo.htmlacademy.ru/" method="get">
```
method="post" скорее

talequale commented on index.html:
```diff
@@ -5,8 +5,153 @@
+<h3 class="distribution-title">Подпишитесь на рассылку</h3>
+<p class="distribution-description">Только полезная информация и никакого спама, честное бойскаутское!</p>
+<form class="distribution-email" action="https://echo.htmlacademy.ru/" method="get">
+<input type="text" name="email" placeholder="Ваш e-mail">
```
для полей ввода электронной почты используется тип email, а не text

talequale commented on index.html:
```diff
@@ -5,8 +5,153 @@
+  <a href="https://htmlacademy.ru"><img class="footer-logo" src="#" alt="Логотип HTMLAcademy"></a>
+</section>
+    </footer>
+<p class="visually-hidden">Репозиторий создан для обучения на профессиональном онлайн‑курсе «<a href="https://htmlacademy.ru/intensive/htmlcss">HTML и CSS. Профессиональная вёрстка сайтов</a>».</p>
```
эту строку удалить нужно

talequale commented on catalog.html:
```diff
@@ -0,0 +1,196 @@
+    <a href="index.html" alt="Ссылка на главную страницу"><img class="house-picture" src="" alt="картнка домика"></a>
+    <span>&gt;</span>
+    <a>Гостиницы</a>
+  </div>
```
весь этот блок - "хлебные крошки", можно делать списком ссылок

talequale commented on catalog.html:
```diff
@@ -0,0 +1,196 @@
+  <h1 class="cataloge-filter-title">Гостиницы Седоны </h1>
+  <div class="extra-link">
+    <a href="index.html" alt="Ссылка на главную страницу"><img class="house-picture" src="" alt="картнка домика"></a>
+    <span>&gt;</span>
```
это декоративный элемент, его не надо размечать

talequale commented on catalog.html:
```diff
@@ -0,0 +1,196 @@
+  </ul>
+
+
+  <span class="filtre-price prcice">Стоимость, &#8381;:</span>
```
это такой же элемент, как "Инфраструктура" и "Тип Жилья", которые у тебя сделаны заголовками  h3. логично и тут заголовок сделать, а не спан

talequale commented on catalog.html:
```diff
@@ -0,0 +1,196 @@
+  <span class="filtre-price prcice">Стоимость, &#8381;:</span>
+  <input class="price-filter" type="text" value="0 от">
+  <input class="price-filter" type="text" value="9000 до">
+  <input class="filter-range" type="range" name="range"> <!--- Линия ползунок --->
```
input type="range" не стилизуется кроссбраузерно, кроме того, он бывает только с одним ползунком, а не с двумя. поэтому потом мы будем верстать кастомный элемент (предполагается, что программист напишет скрипт для того, чтоб ползунок был рабочим. мы заставлять его работать не будем, просто сверстаем)

talequale commented on catalog.html:
```diff
@@ -0,0 +1,196 @@
+  <input class="price-filter" type="text" value="9000 до">
+  <input class="filter-range" type="range" name="range"> <!--- Линия ползунок --->
+  <button class="use" type="submit" value="Применить">Применить</button>
+  <button class="remove" type="submit" value="Применить">Сбросить</button>
```
это button type="reset"
value кнопкам кстати не обязательно писать

talequale commented on catalog.html:
```diff
@@ -0,0 +1,196 @@
+      <option class="control">Сначала дорогие</option>
+      <option class="control">Сначала популярные</option>
+    </select>
+    <a href="#" class="button button-light">
```
класс button-light предполагает, что это какой-то модифицированный вариант класса button (кстати, этот класс button у тебя уже используется для социальных кнопок, которые не очень похожи на эти кнопки. разные элементы лучше не делать одинаковыми классами, запутаешься. назови эти кнопки grid-button, например). а на самом деле на первой кнопке показано состояние выбранного элемента (то есть в данный момент результаты поиска показаны карточками). лучше для показа таких состояний использовать какие-то адекватные классы типа active, current и т д

talequale commented on catalog.html:
```diff
@@ -0,0 +1,196 @@
+  <ul class="product-list">
+<li class="product-card">
+  <img class="product-card-image" src="" alt="Амара Резорт & СПА">
+  <h3 class="product-card-title">Amara Resort & Spa</h3>
```
судя по наличию состояний в стайлгайде для этого элемента, он должен быть ссылкой. это и логично, как-то со страницы с результатами поиска мы должны попасть на страницу конкретного отеля. с одной стороны, у нас есть кнопка "Подробнее", но обычно ссылками делают всё, на что пользователь может кликать. в нашем случае это заголовок и картинка

talequale commented on catalog.html:
```diff
@@ -0,0 +1,196 @@
+<li class="product-card">
+  <img class="product-card-image" src="" alt="Амара Резорт & СПА">
+  <h3 class="product-card-title">Amara Resort & Spa</h3>
+  <Span class="product-item">Гостиница</Span>
```
имена тегов пишутся со строчной буквы

talequale commented on catalog.html:
```diff
@@ -0,0 +1,196 @@
+  <h3 class="product-card-title">Amara Resort & Spa</h3>
+  <Span class="product-item">Гостиница</Span>
+  <Span class="product-item">от 4000 &#8381;</Span>
+  <a class="product-description">Подробнее</a>
+  <a class="product-selected">в избранное</a>
```
у ссылок должен быть атрибут href

talequale commented on catalog.html:
```diff
@@ -0,0 +1,196 @@
+  <a href="https://htmlacademy.ru"><img class="footer-logo" src="#" alt="Логотип HTMLAcademy"></a>
+</section>
+    </footer>
+<p class="visually-hidden">Репозиторий создан для обучения на профессиональном онлайн‑курсе «<a href="https://htmlacademy.ru/intensive/htmlcss">HTML и CSS. Профессиональная вёрстка сайтов</a>».</p>
```
эту строчку надо удалить

talequale commented on catalog.html:
```diff
@@ -0,0 +1,196 @@
+</section>
+    </footer>
+<p class="visually-hidden">Репозиторий создан для обучения на профессиональном онлайн‑курсе «<a href="https://htmlacademy.ru/intensive/htmlcss">HTML и CSS. Профессиональная вёрстка сайтов</a>».</p>
+  </body>
+</html>
+  </body>
+</html>
```
дважды закрываются теги body и html

talequale commented on index.html:
```diff
@@ -5,8 +5,153 @@
+
+  <ul class="advantages-list">
+    <li class="advantages-item">
+      <img class="advantages-picture" src="" alt="Картинка дома"
```
тег img не закрыт

talequale commented on index.html:
```diff
@@ -5,8 +5,153 @@
+
+  <ul class="advantages-list">
+    <li class="advantages-item">
+      <img class="advantages-picture" src="" alt="Картинка бургера"
```
тег img не закрыт

talequale commented on index.html:
```diff
@@ -5,8 +5,153 @@
+
+  <ul class="advantages-list">
+    <li class="advantages-item">
+      <img class="advantages-picture" src="" alt="Картинка сувенира"
```
тег img не закрыт

talequale commented on catalog.html:
```diff
@@ -0,0 +1,196 @@
+<section class="catalog-filter">
+  <h1 class="cataloge-filter-title">Гостиницы Седоны </h1>
+  <div class="extra-link">
+    <a href="index.html" alt="Ссылка на главную страницу"><img class="house-picture" src="" alt="картнка домика"></a>
```
у ссылок не бывает атрибута alt, он бывает только у картинок

talequale commented on catalog.html:
```diff
@@ -0,0 +1,196 @@
+  </div>
+  <h2 class="visually-hidden">Фильтр</h2>
+  <h3 class="filter-title">Инфраструктура</h3>
+  <form class="catalog-filter" action="https://echo.htmlacademy.ru/" method="get"></form>
```
тег form открыт и сразу закрыт, то есть он пустой, а весь контент, который идёт за ним, не лежит в теге form

talequale reviewed (commented):
1) в обоих файлах проблемы с индентацией, код очень сложно читать, так как не видна вложенность. нужно с этим разобраться. везде должен быть отступ, равный двум пробелам. вложенные теги должны быть сдвинуты относительно родительских на один отступ вправо
сейчас код выглядит так:
![image](https://user-images.githubusercontent.com/11165551/192356783-491cf293-101b-4fe8-9031-34541cc1295b.png)

а, для сравнения, тот же участок кода должен выглядеть так:
![image](https://user-images.githubusercontent.com/11165551/192356980-ca826b61-9c9a-4212-b049-ee1f668c79c2.png)

и так везде нужно поправить оформление кода в соответствии с кодгайдом https://up.htmlacademy.ru/htmlcss/35/criteries#PROJ-01

2) есть ошибки валидации, проверить html на валидность можно тут https://validator.w3.org/nu/#textarea . все ошибки валидатора надо исправить

AndreyKaliushka committed:
Добавляет исправления

Привет! Я добавил все исправления (надеюсь).
По валидатору оба документа прогнал, ошибки все убрал, предупреждения тоже.
Сейчас валидатор проходит как "чистый"

AndreyKaliushka committed:
Добавляет исправления

Привет! Я добавил все исправления (надеюсь).
По валидатору оба документа прогнал. Все ошибки и предупреждения исправил.

talequale commented on index.html:
```diff
@@ -1,12 +1,154 @@
   <body>
+    <header class="page-header">
+      <nav class="navigation">
+        <img class="logo" src="" alt="Логотип Седона">
```
логотип это ссылка на обеих страницах, на главной это может быть ссылка без атрибута href

talequale commented on index.html:
```diff
@@ -1,12 +1,154 @@
+            <a class="navigation-link">Лайки<sup>12</sup></a>
+          </li>
+          <li class="navigation-item">
+            <input class="header-button" type="button" name="ХОЧУ СЮДА" value="ХОЧУ СЮДА!">
```
для разметки кнопок используется тег button, а не input type="button" (он использовался, когда не было button, а теперь в этом качестве устарел)

talequale commented on index.html:
```diff
@@ -1,12 +1,154 @@
+        <img class="main-logo" src="" alt="Логотип Седона">
+      </div>
+      <section class="reason">
+        <h2 class="reason first">Седона — небольшой городок в Аризоне, заслуживающий большего!</h2>
```
тут наверное дефис пропущен. вряд ли у обёртки блока и у заголовка совпадает класс (reason)

talequale commented on index.html:
```diff
@@ -1,12 +1,154 @@
+      <section class="interested">
+        <h3 class="interested-title">Заинтересовались?</h3>
+        <p class="interested-description">Укажите предполагаемые даты поездки, и мы покажем вам лучшие предложения гостиниц в Седоне</p>
+        <input class="interested-button" type="button" name="ПОИСК ГОСТИНИЦЫ В седоне" value="ПОИСК ГОСТИНИЦЫ В седоне">
```
тут надо использовать тег button либо тег a. тег input для разметки кнопок уже не используется

talequale commented on catalog.html:
```diff
@@ -0,0 +1,226 @@
+        <h1 class="cataloge-filter-title">Гостиницы Седоны </h1>
+        <ul class="extra-link">
+          <li><a href="index.html"><img class="house-picture" src="" alt="картнка домика"></a></li>
+            <!--<span>&gt;</span> Этот элемент декаративный?-->
```
да, декоративный, в разметку добавлять не нужно

talequale commented on catalog.html:
```diff
@@ -0,0 +1,226 @@
+          </ul>
+
+
+          <h3 class="filtre-price prcice">Стоимость, &#8381;:</h3>
```
опечатка видимо, prcice вместо price

talequale commented on catalog.html:
```diff
@@ -0,0 +1,226 @@
+            <option class="control">Сначала дорогие</option>
+            <option class="control">Сначала популярные</option>
+          </select>
+          <a href="#" class="active-button">
+            <span class="visually-hidden">Показать карточки</span>
+          </a>
+          <a class="active-button" href="#">
+            <span class="visually-hidden">Показать слайдшоу </span>
+          </a>
+          <a class="active-button" href="#">
+            <span class="visually-hidden">Показать список</span>
+          </a>
```
не все же кнопки активные, на макете только первая показана активной. я имела в виду, что у всех кнопок должен быть общий класс, а у активной ещё и дополнительный 

```
<a class="grid-button grid-button-active" href="#">Показать карточки</a>
<a class="grid-button" href="#">Показать слайдшоу</a>
<a class="grid-button" href="#">Показать список</a>
```

talequale commented on catalog.html:
```diff
@@ -0,0 +1,226 @@
+      <section>
+        <ul class="product-list">
+          <li class="product-card">
+            <a href="#">
+              <img class="product-card-image" src="" alt="Амара Резорт & СПА">
+            </a>
+            <a href="#">
+              <h3 class="product-card-title">Amara Resort & Spa</h3>
+            </a>
```
класс всегда лучше давать внешнему элементу, а не внутреннему (либо и тому, и другому). потому что по каскаду нам внутренний элемент доступен от внешнего, а от класса внутреннего добраться стилями до внешнего мы не можем

AndreyKaliushka committed:
Добавляет правки

гитхабу определенно не хватает смайликов в чате

AndreyKaliushka reviewed (commented):
Скажи, пожалуйста, когда стал выглядеть читабельным?

talequale commented on index.html:
```diff
@@ -1,12 +1,153 @@
+            <a class="navigation-link">Лайки<sup>12</sup></a>
+          </li>
+          <li class="navigation-item">
+            <button class="header-button" type="submit" name="header-button" value="Хочу сюда">Хочу сюда</button>
```
type="submit" говорит о том, что кнопка отправляет данные из формы. но тут нет формы, и никакие данные не отправляются. поэтому у этой кнопки type="button"

talequale commented on index.html:
```diff
@@ -1,12 +1,153 @@
   <body>
+    <header class="page-header">
+      <nav class="navigation">
+        <a href=""><img class="logo" src="" alt="Логотип Седона"></a>
```
я имела в виду, что можно вообще атрибут href убрать, тогда ссылка будет неактивной (потому что мы и так на главной странице), но при этом мы сохраним консистентность кода на всех страницах (везде одинаковые теги для логотипа использованы).
если оставляем href, то и его значение (index.html) надо оставить

talequale commented on index.html:
```diff
@@ -1,12 +1,153 @@
+      <section class="interested">
+        <h3 class="interested-title">Заинтересовались?</h3>
+        <p class="interested-description">Укажите предполагаемые даты поездки, и мы покажем вам лучшие предложения гостиниц в Седоне</p>
+        <button class="interested-button" type="submit" name="interested-button" value="ПОИСК ГОСТИНИЦЫ В седоне">Поиск гостиницы в седоне</button>
```
это button type="button"

talequale commented on catalog.html:
```diff
@@ -0,0 +1,227 @@
+      <section>
+        <ul class="product-list">
+          <li class="product-card">
+            <a class="product-card-image" href="#">
+              <img class="product-card-image" src="" alt="Амара Резорт & СПА">
+            </a>
```
у двух вложенных разных элементов не может быть одинаковых классов. ссылка это ссылка, картинка это картинка. классы должны это отражать

talequale commented on catalog.html:
```diff
@@ -0,0 +1,227 @@
+            <a class="product-card-image" href="#">
+              <img class="product-card-image" src="" alt="Амара Резорт & СПА">
+            </a>
+            <a href="#">
+              <h3 class="product-card-title">Amara Resort & Spa</h3>
+            </a>
```
класс для внешнего элемента нужен не только обёртке картинки, заголовка это тоже касалось

talequale commented:
> Скажи, пожалуйста, когда стал выглядеть читабельным?

ты про отступы? да, стало нормально

talequale commented on catalog.html:
```diff
@@ -0,0 +1,227 @@
+        <form class="catalog-filter" action="https://echo.htmlacademy.ru/" method="get">
+          <h3 class="filter-title">Инфраструктура</h3>
+          <ul class="filter-item">
+            <li>
+              <input class="checkbox-button" type="checkbox" name="pool" checked> Бассейн
+            </li>
+            <li>
+              <input class="checkbox-button" type="checkbox" name="park" checked> Парковка
+            </li>
+            <li>
+              <input class="checkbox-button" type="checkbox" name="wi-fi"> WI-FI
```
не обратила внимания на проблему с формой. и тут, и у радиокнопок отстутсвуют ассоциированные с полями лейблы (тег label). label и input связываются одинаковыми атрибутами for и id соответственно

talequale commented on catalog.html:
```diff
@@ -0,0 +1,227 @@
+
+
+          <h3 class="filtre-price price">Стоимость, &#8381;:</h3>
+          <input class="price-filter" type="text" value="0 от">
+          <input class="price-filter" type="text" value="9000 до">
```
"от" и "до" это подписи полей (label), а не их значения

AndreyKaliushka committed:
Добавляет новые-новые исправления

talequale commented on catalog.html:
```diff
@@ -0,0 +1,237 @@
+            <a class="navigation-link">Лайки<sup>12</sup></a>
+          </li>
+          <li class="navigation-item">
+            <a class="navigation-link-button" href="#">Хочу сюда</a>
```
нужно одинаково сделать на обеих страницах. почему там button, а тут ссылка?

talequale commented on catalog.html:
```diff
@@ -0,0 +1,237 @@
+          <h3 class="filter-title">Тип Жилья</h3>
+
+          <ul class="filter-item">
+            <li>
+              <input class="radio-button" id="hotel" type="radio" name="hotel" checked>
+              <label class="filter-label" for="hotel">Гостиница</label>
+            </li>
+            <li>
+              <input class="radio-button" id="hostel" type="radio" name="hostel">
+              <label class="filter-label" for="hostel">Мотель</label>
+            </li>
+            <li>
+              <input class="radio-button" id="flats" type="radio" name="flats">
+              <label class="filter-label" for="flats">Апартаменты</label>
+            </li>
```
чтобы группа радиокнопок переключалась, у них должен быть одинаковый атрибут name
